orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/AbstractRedisExecutionRepository.java

```diff
@@ -0,0 +1,343 @@
+      map.put(prefix + "lastModified", (stage.getLastModified() != null ? mapper.writeValueAsString(stage.getLastModified()) : null));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Failed converting stage to json", e);
+    }
+    return map;
+  }
+
+  private void deleteInternal(JedisCommands c, ExecutionType type, String id) {
+    String key = format("%s:%s", type.toString(), id);
+    try {
+      String application = c.hget(key, "application");
+      String appKey = appKey(type, application);
+      c.srem(appKey, id);
+
+      if (type == PIPELINE) {
+        String pipelineConfigId = c.hget(key, "pipelineConfigId");
+        c.zrem(executionsByPipelineKey(pipelineConfigId), id);
+      }
+    } catch (ExecutionNotFoundException ignored) {
+      // do nothing
+    } finally {
+      c.del(key);
+      c.srem(alljobsKey(type), id);
+    }
+  }
+
+  static String alljobsKey(ExecutionType type) {
+    return format("allJobs:%s", type.toString());
+  }
+
+  static String appKey(ExecutionType type, String app) {
+    return format("%s:app:%s", type.toString(), app);
+  }
+
+  static String pipelineKey(String id) {
+    return format("pipeline:%s", id);
+  }
+
+  static String orchestrationKey(String id) {
+    return format("orchestration:%s", id);
+  }
+
+  protected static String executionsByPipelineKey(String pipelineConfigId) {
+    String id = pipelineConfigId != null ? pipelineConfigId : "---";
+    return format("pipeline:executions:%s", id);
+  }
+
+  private String fetchKey(String id) {
+    String key = redisClientDelegate.withCommandsClient(c -> {
+      if (c.exists(pipelineKey(id))) {
+        return pipelineKey(id);
+      } else if (c.exists(orchestrationKey(id))) {
+        return orchestrationKey(id);
+      }
+      return null;
+    });
+
+    if (key == null && previousRedisClientDelegate.isPresent()) {
+      key = redisClientDelegate.withCommandsClient(c -> {
+        if (c.exists(pipelineKey(id))) {
+          return pipelineKey(id);
+        } else if (c.exists(orchestrationKey(id))) {
+          return orchestrationKey(id);
+        }
+        return null;
+      });
+    }
+
+    if (key == null) {
+      throw new ExecutionNotFoundException("No execution found with id " + id);
+    }
+    return key;
+  }
+
+  RedisClientDelegate getRedisDelegateForId(String id) {
+    if (StringUtils.isEmpty(id)) {
+      return redisClientDelegate;
+    }
+
+    final RedisClientDelegate[] delegate = {null};
+
+    redisClientDelegate.withCommandsClient(c -> {
+      if (c.exists(id)) {
+        delegate[0] = redisClientDelegate;
+      }
+    });
+
+    if (delegate[0] == null && previousRedisClientDelegate.isPresent()) {
+      previousRedisClientDelegate.get().withCommandsClient(c -> {
+        if (c.exists(id)) {
+          delegate[0] = previousRedisClientDelegate.get();
+        }
+      });
+    }
+
+    return (delegate[0] == null) ? redisClientDelegate : delegate[0];
```

Comment:
I would consider changing the `delegate` from a `final` and skipping the array usage. I'm not sure it's buying us anything since the value of `delegate` is always only one, and that value isn't final.

Reply:
done, will be addressed in the next push.